systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -87,20 +99,70 @@ public class MigrationST extends AbstractST {
+    }
+
+    /**
+     * @description This testcase is focused on rollback process after first part of the migration from ZK to KRaft is done.
+     * It goes through whole process, together with checking that message transmission throughout the test will not be
+     * disrupted.
+     *
+     * @steps
+     *  1. - Deploys Kafka resource (with enabled NodePools and KRaft set to disabled) with Broker NodePool
+     *  2. - Deploys Controller NodePool - it's created here, so we will firstly delete KafkaTopics in our ResourceManager;
+     *       the Pods will not be created until the migration starts
+     *  3. - Creates topics for continuous and immediate message transmission, TLS user
+     *  4. - Starts continuous producer & consumer
+     *  5. - Does immediate message transmission
+     *  6. - Starts the migration
+     *  7. - Annotates the Kafka resource with strimzi.io/kraft:migration
+     *  8. - Controllers will be created and moved to RUNNING state
+     *  9. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaftMigration
+     *  10. - Waits for first rolling update of Broker Pods - bringing the Brokers to DualWrite mode
+     *  11. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaftDualWriting
+     *  12. - Waits for second rolling update of Broker Pods - removing dependency on ZK
+     *  13. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaftPostMigration
+     *  14. - Creates a new KafkaTopic and checks both ZK and KRaft metadata for presence of the KafkaTopic
+     *  15. - Does immediate message transmission to the new KafkaTopic
+     *  16. - Rolling back the migration - annotates the Kafka resource with strimzi.io/kraft:rollback
+     *  17. - Waits for rolling update of Broker Pods - adding dependency on ZK back
+     *  18. - Checks that Kafka CR has .status.kafkaMetadataState set to KRaftDualWriting
+     *  19. - Deletes the Controller NodePool
+     *  20. - Finishes rollback - annotates the Kafka resource with strimzi.io/kraft:disabled
+     *  21. - Waits for rolling update of Broker Pods - rolling back from DualWrite mode to ZooKeeper
+     *  22. - Checks that Kafka CR has .status.kafkaMetadataState set to ZooKeeper
+     *  23. - Checks that __cluster_metadata topic doesn't exist in Kafka Brokers
+     *  24. - Waits until continuous clients are finished successfully
+     *
+     * @usecase
+     *  - zk-to-kraft-migration
+     */
+    @IsolatedTest
+    void testRollbackDuringMigration(ExtensionContext extensionContext) {
+        TestStorage testStorage = new TestStorage(extensionContext, TestConstants.CO_NAMESPACE);
+
+        setupMigrationTestCase(extensionContext, testStorage);
+        doFirstPartOfMigration(extensionContext, testStorage);
+        doRollback(testStorage);
+    }
+
+    private void setupMigrationTestCase(ExtensionContext extensionContext, TestStorage testStorage) {
+        postMigrationTopicName = testStorage.getTopicName() + "-midstep";
```

Comment:
we are using `-post-migration` in the current codebase for migration ST as the variable name suggests

Reply:
Yeah good catch, I missed this one when I was copying the changes between the branches. Thanks!